src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/rca/RcaController.java

```diff
@@ -268,13 +268,35 @@ private void readRcaEnabledFromConf() {
             String nextLine = sc.nextLine();
             rcaEnabled = Boolean.parseBoolean(nextLine);
           } catch (IOException e) {
-            LOG.error("Error reading file '{}': {}", filePath.toString(), e.getMessage());
+            LOG.error("Error reading file '{}': {}", filePath.toString(), e);
             e.printStackTrace();
             rcaEnabled = rcaEnabledDefaultValue;
           }
         });
   }
 
+  private void readAndUpdateMutesRcasDuringStart() {
+    try {
+      /* We have an edge case where both `readAndUpdateMutesRcas()` and `readAndUpdateMutesRcasDuringStart()`
+       * can try to update the muted Rca list back to back, reading rca.conf twice. This will happen when rca
+       * was turned off and then on.
+       *
+       * <p> `readAndUpdateMutesRcasDuringStart()` should only be read at the start of the process, when
+       * RCA graph is not constructed and we cannot validate the new new muted RCAs. For any other update
+       * to the muted list, the periodic rca.conf update checker will take care of it.
+       *
+       */
+      if (lastModifiedTimeInMillisInMemory == 0) {
+        Set<String> rcasForMute = new HashSet<>(rcaConf.getMutedRcaList());
+        rcasForMute.forEach(mutedRca -> Stats.getInstance().addToMutedGraphNodes(mutedRca));
```

Comment:
Will it make sense to rewrite `updateMutedGraphNodes` as do two things:
- clear()
- addAll(nodeNames)

And just call that here instead of iterating through `rcasForMute` ?

Reply:
Yes, we can use the `updateMutedGraphNodes()` as is. Updating the code for the same.